github-adapter/src/main/java/be/xplore/githubmetrics/githubadapter/mappingclasses/GHSelfHostedRunner.java

```diff
@@ -3,28 +3,49 @@
 import be.xplore.githubmetrics.domain.selfhostedrunner.OperatingSystem;
 import be.xplore.githubmetrics.domain.selfhostedrunner.SelfHostedRunner;
 import be.xplore.githubmetrics.domain.selfhostedrunner.SelfHostedRunnerStatus;
+import be.xplore.githubmetrics.githubadapter.config.GithubProperties;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.text.MessageFormat;
+import java.util.List;
+import java.util.Locale;
 
 public record GHSelfHostedRunner(
         String os,
         String status,
         boolean busy
 ) {
-    public SelfHostedRunner getRunner() {
-        return new SelfHostedRunner(this.getOs(), this.getStatus());
+    private static final Logger LOGGER = LoggerFactory.getLogger(GHSelfHostedRunner.class);
+
+    public SelfHostedRunner getRunner(GithubProperties.Parsing.SelfHostedRunnerOsKeywords keywords) {
+        return new SelfHostedRunner(this.getOs(keywords), this.getStatus());
     }
 
-    private OperatingSystem getOs() {
-        return switch (this.os) {
-            case "linux" -> OperatingSystem.LINUX;
-            case "windows" -> OperatingSystem.WINDOWS;
-            case "macOS" -> OperatingSystem.MAC_OS;
-            default -> throw new IllegalStateException(MessageFormat.format(
-                    "The string {0} could not be parsed to the OperatingSystem enum.",
+    private OperatingSystem getOs(GithubProperties.Parsing.SelfHostedRunnerOsKeywords keywords) {
+        LOGGER.error(this.os);
+        var osEnum = OperatingSystem.UNKNOWN;
+        if (osContainsAny(keywords.linuxKeywords())) {
+            osEnum = OperatingSystem.LINUX;
+        } else if (osContainsAny(keywords.windowsKeywords())) {
+            osEnum = OperatingSystem.WINDOWS;
+        } else if (osContainsAny(keywords.macosKeywords())) {
+            osEnum = OperatingSystem.MAC_OS;
+        } else {
+            LOGGER.warn("""
+                                  Unknown operating system encountered. Please update the keywords so that we are able to match the os to one of the options.
+                                  Consider that all keywords and os's are treated as lowercase. Os was {}
+                            """,
                     this.os
-            ));
-        };
+            );
```

Comment:
Could this be better as multiple logs instead of text block

Reply:
done